internal/page/certificateprovider/onelogin_identity_details.go

```diff
@@ -0,0 +1,60 @@
+package certificateprovider
+
+import (
+	"net/http"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/actor"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+)
+
+type oneloginIdentityDetailsData struct {
+	App                 page.AppData
+	Errors              validation.List
+	CertificateProvider *actor.CertificateProviderProvidedDetails
+}
+
+func OneloginIdentityDetails(tmpl template.Template, certificateProviderStore CertificateProviderStore, lpaStoreResolvingService LpaStoreResolvingService, lpaStoreClient LpaStoreClient) page.Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request) error {
+		certificateProvider, err := certificateProviderStore.Get(r.Context())
+		if err != nil {
+			return err
+		}
+
+		data := &oneloginIdentityDetailsData{
+			App:                 appData,
+			CertificateProvider: certificateProvider,
+		}
+
+		if r.Method == http.MethodPost {
+			lpa, err := lpaStoreResolvingService.Get(r.Context())
+			if err != nil {
+				return err
+			}
+
+			if certificateProvider.CertificateProviderIdentityConfirmed(
+				lpa.CertificateProvider.FirstNames,
+				lpa.CertificateProvider.LastName,
+			) {
+				certificateProvider.Tasks.ConfirmYourIdentity = actor.TaskCompleted
+
+				err = certificateProviderStore.Put(r.Context(), certificateProvider)
+				if err != nil {
+					return err
+				}
+
+				if err := lpaStoreClient.SendCertificateProviderConfirmIdentity(r.Context(), lpa.LpaUID, certificateProvider); err != nil {
```

Comment:
I think this should move back, even though it makes the other handler bigger, because that is the point identity is confirmed